Tools/clinic/clinic.py

```diff
@@ -4325,34 +4340,34 @@ def directive_output(self, command_or_name, destination=''):
             fail("Invalid command / destination name " + repr(command_or_name) + ", must be one of:\n  preset push pop print everything " + " ".join(fd))
         fd[command_or_name] = d
 
-    def directive_dump(self, name):
+    def directive_dump(self, name: str) -> None:
         self.block.output.append(self.clinic.get_destination(name).dump())
 
-    def directive_printout(self, *args):
+    def directive_printout(self, *args) -> None:
```

Comment:
All args in the `*args` tuple must be `str`s, as they're immediately passed to `' '.join()`

```suggestion
    def directive_printout(self, *args: str) -> None:
```